client_v0_test.go

```diff
@@ -153,12 +154,65 @@ func TestUserAgentV0(t *testing.T) {
 	client.accessTokenGetter = client.getAccessToken
 	client.parameterGetter = client.getQueryParams
 
-	_, _ = client.Query(context.TODO(), server.URL, "dummy", "SELECT 1", map[string]string{})
+	_, _ = client.Query(context.TODO(), server.URL, "SELECT 1", map[string]string{}, func(key, value string) {})
 	if userAgentHeader != userAgentValue {
 		t.Errorf("Did not set User-Agent value correctly on a query request")
 	}
 }
 
+// TestProtocolVersion tests that protocol version is correctly set on request
+func TestProtocolVersionV0(t *testing.T) {
+	var protocolVersionValue = ""
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		protocolVersionValue = r.Header.Get(protocolVersionHeader)
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+	prepareEnvVariablesForTest(t, server)
+	var client = &ClientImplV0{
+		BaseClient{ClientID: "ClientID@firebolt.io", ClientSecret: "password", ApiEndpoint: server.URL},
+	}
+	client.accessTokenGetter = client.getAccessToken
+	client.parameterGetter = client.getQueryParams
+
+	_, _ = client.Query(context.TODO(), server.URL, "SELECT 1", map[string]string{}, func(key, value string) {})
+	if protocolVersionValue != protocolVersion {
+		t.Errorf("Did not set Protocol-Version value correctly on a query request")
+	}
+}
+
+func TestUpdateParametersV0(t *testing.T) {
+	var newDatabaseName = "new_database"
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == UsernamePasswordURLSuffix {
+			_, _ = w.Write(getAuthResponseV0(10000))
+		} else {
+			w.Header().Set(updateParametersHeader, fmt.Sprintf("%s=%s", "database", newDatabaseName))
+			w.WriteHeader(http.StatusOK)
+		}
+	}))
+	defer server.Close()
+	prepareEnvVariablesForTest(t, server)
+	var client = &ClientImplV0{
+		BaseClient{ClientID: "ClientID@firebolt.io", ClientSecret: "password", ApiEndpoint: server.URL},
+	}
+	client.accessTokenGetter = client.getAccessToken
+	client.parameterGetter = client.getQueryParams
+
+	params := map[string]string{
+		"database": "db",
+	}
+	_, err := client.Query(context.TODO(), server.URL, "SELECT 1", params, func(key, value string) {
+		params[key] = value
+	})
+	if err != nil {
+		t.Errorf("Error during query execution with update parameters header in response %s", err)
+	}
+	if params["database"] != newDatabaseName {
+		t.Errorf("Did not set Update-Parameters value correctly")
```

Comment:
I'd make this message clearer. It is thrown when the returned database is not as expected. Should we probably say "Database is not set correctly. Expected {newDatabaseName} but was {params["database"]}"